src/Runtime/ServiceInstantiator.php

```diff
@@ -0,0 +1,204 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drush\Runtime;
+
+use Drush\Log\Logger;
+use League\Container\Container;
+use Drush\Drush;
+use Symfony\Component\Console\Application;
+use Composer\Autoload\ClassLoader;
+use Drush\Command\DrushCommandInfoAlterer;
+
+use Psr\Container\ContainerInterface;
+use League\Container\Container as DrushContainer;
+use Symfony\Component\Yaml\Yaml;
+use Robo\Robo;
+
+/**
+ * Use Dependency Injection Container to instantiate services.
+ */
+class ServiceInstantiator
+{
+    protected ContainerInterface $drushServicesContainer;
+    protected array $tags = [];
+
+    public function __construct(protected ContainerInterface $container, protected ContainerInterface $drushContainer)
+    {
+        $this->drushServicesContainer = new DrushContainer();
```

Comment:
Services created in a drush.services.yml file are stored in this container so that they can be referenced as needed e.g. as the parameter to some other service or command also in a drush.services.yml file. Usually, only services in the same file can reference each other, as there is no other way to ensure the initialization order of Drush services.

Reply:
Maybe with a little extra work, we could use the features of league container to lazy-initialize the commandfiles. I'm not sure if it's necessary to allow cross-Drush-service-file references, though. Since Drush services can reference Drupal services, it would be preferable to just make a Drupal service for any shared service needs a Drush command might have.

Reply:
Is this a third container? I have no idea if there is a better way but its at least a thing that will raise an eyebrow.

1. Original Drush container
2. New container containing Drupal service data
3. Drupal container.

Reply:
We could re-use the drupal container or the drush container or just put it in an array. The scope is only during instantiation; it is not needed thereafter.